src/cs/Ssh.Tcp/Services/PortForwardingService.cs

```diff
@@ -468,7 +495,16 @@ public PortForwardingService(SshSession session)
 			forwardedPort,
 			cancellation).ConfigureAwait(false);
 
-		return new SshStream(channel);
+		var forwardedStream = await OnForwardedPortConnectingAsync(
+			forwardedPort, isIncoming: false, new SshStream(channel), cancellation)
+			.ConfigureAwait(false);
+		if (forwardedStream == null)
+		{
+			throw new SshChannelException(
```

Comment:
Should we close the SSH channel in this case? Also if the `ForwardedPortConnecting` handler throws.

Reply:
Updated.